src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -180,11 +180,20 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
             }
 
             var constructorArguments = analyzedArguments.ConstructorArguments;
-            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.ToImmutableAndFree();
+
+            // If the arguments didn't have a natural type, they're not constants, and so would already have diagnostics from the rest of binding.
+            var ignoredDiagnostics = new DiagnosticBag();
+            ImmutableArray<BoundExpression> boundConstructorArguments = constructorArguments.Arguments.SelectAsArray(
+                (arg, thisAndDiagnostics) => thisAndDiagnostics.@this.BindToNaturalType(arg, thisAndDiagnostics.ignoredDiagnostics),
```

Comment:
>BindToNaturalType [](start = 70, length = 17)

```BindToTypeForErrorRecovery```?

Reply:
We also have a ```BuildArgumentsForErrorRecovery```, see if we can use that helper. 

---
In reply to: [529823025](https://github.com/dotnet/roslyn/pull/49534#discussion_r529823025) [](ancestors = 529823025)